src/biz/collab/ops.rs

```diff
@@ -804,5 +973,74 @@ pub async fn list_database_row_details(
     })
     .collect::<Vec<AFDatabaseRowDetail>>();
 
-  Ok(database_row_details)
+  // Fill in the document content if requested and exists
+  if with_doc {
+    let doc_id_by_row_id = db_row_details
+      .iter()
+      .filter(|row| row.has_doc)
+      .flat_map(|row| {
+        row.id.parse::<Uuid>().ok().map(|row_uuid| {
+          (
+            row.id.clone(),
+            meta_id_from_row_id(&row_uuid, RowMetaKey::DocumentId),
+          )
+        })
+      })
+      .collect::<HashMap<_, _>>();
+
+    let query_db_docs = doc_id_by_row_id
+      .values()
+      .map(|doc_id| QueryCollab {
+        object_id: doc_id.to_string(),
+        collab_type: CollabType::Document,
+      })
+      .collect::<Vec<_>>();
+
+    let mut query_res = collab_storage
+      .batch_get_collab(&uid, &workspace_uuid_str, query_db_docs, true)
+      .await;
+
+    for row_detail in &mut db_row_details {
```

Comment:
Would it be better to create a function for the following calls? This would allow you to eliminate many match statements and take advantage of the Result type for error handling.

Reply:
have refactored this part into a function